### DISCUSSION TITLE
"🐍" in prompt without a version and without having Python installed

### DISCUSSION THREAD

**Discussion by l0b0** [starship/starship#6380] [Q&A] [answered]
I *think* this is a fairly new issue. When I start a new terminal, it comes up with this prompt :

```console
~ via 🐍 
❯ printf '%q\n' "$PS1"
$'\n\\[\E[1;36m\\]~\\[\E[0m\\] via \\[\E[1;33m\\]🐍 \\[\E[0m\\]\n\\[\E[1;32m\\]❯\\[\E[0m\\] '
```

But I don't have Python installed:

```console
~ via 🐍 
❯ python
The program 'python' is currently not installed. It is provided by
several packages. You can install it by typing one of the following:
[omitted]
```

Is there some way to inspect what Starship thinks is happening?

Running Starship 1.21.1 with GNOME Console on NixOS.

**Comment by davidkna** [comment, marked as answer]
If the python version cannot be determined (including if the executable is missing) the symbol will still be shown without the version.  All modules that display version numbers behave in this manner. The behaviour can be avoided in this and other modules by adjusting the module format:
```toml
[python]
format = '(via [${symbol}${pyenv_prefix}${version} (\($virtualenv\) )]($style))'
```

As to why the module shows, there might be a `.py` file in the home folder.

**Reply by l0b0** [reply]
Hah, yes, there was a .py file there by accident.
